### ISSUE TITLE
Wrong types for `assertIsNotNumeric`

### ISSUE THREAD

**Issue by ZebulanStanphill** [phpstan/phpstan-phpunit#124]
The following TestCase will cause PHPStan to report a false positive.

```php
<?php declare(strict_types=1);

use PHPUnit\Framework\TestCase;

/**
 * @phpstan-return non-empty-string
 */
function someFunction(): string {
	return '10';
}

final class MyTest extends TestCase {
	public function testGeneration(): void {
		self::assertIsNotNumeric(
			someFunction(),
			'This expectation message is misinterpreted as the "actual" param.'
		);
	}
}
```

The false positive:

> Call to static method PHPUnit\Framework\Assert::assertIsNotNumeric() with non-empty-string and 'This expectation…' will always evaluate to true.

**Comment by ondrejmirtes** [comment]
I don't get it - in my opinion PHPStan is right, that string is not numeric which means the assert will always pass. Which means you can delete that line, the test is useless.

**Comment by ZebulanStanphill** [comment]
@ondrejmirtes Sorry, that example was simplified from my actual use case. The actual use case involves calling a function that returns a `non-empty-string`. I've updated the example to be slightly closer to the real thing.

Also, as you can see in the error message, it's comparing the **wrong string**.

**Comment by ondrejmirtes** [comment]
I confirm, it's a bug.

**Comment by ondrejmirtes** [comment]
But it's not related to the 2nd arg, it reproduces even without it:

> +'09: Call to static method PHPUnit\Framework\Assert::assertIsNotNumeric() with non-empty-string will always evaluate to true.

**Comment by VincentLanglet** [comment]
I don't think the issue exists anymore
https://phpstan.org/r/c487bfe6-6719-4ef5-b43c-85d76296bcbe

I didn't find anything related to `assertIsNotNumeric` is phpstan-phpunit, so it just rely on the phpdoc
```
@phpstan-assert !numeric $actual
```
from PHPUnit.
